### ISSUE TITLE
Should `SetBlendConstant` take float,float,float,float instead of WGPUColor?

### ISSUE THREAD

**Issue by kainino0x** [webgpu-native/webgpu-headers#378]
`WGPUColor` is set up to use `double` because it can represent all of the possible render pass clear values - it's used as a union of `f32`, `i32`, and `u32`.

However, AFAIK, blending, and therefore `wgpuRenderPassEncoderSetBlendConstant`, never applies to sint/uint type textures, so should we just take four 32-bit `float`s instead? Slightly more efficient.

**Comment by Kangz** [comment]
I checked that all APIs do take 32bit floats.

**Comment by kainino0x** [comment]
Dec 12 meeting:

* KN: Worth simplifying? SetBlendConstant only needs f32, never i32/u32 (WGPUColor uses f64 to represent all three)  
* KN: Came up because some bindings might want to have some kind of union-like f32/i32/u32 for colors. WGPUColor designed to represent a color in a texture, which is not exactly what SetBlendConstant is.
* (discussion of why to do this change)  
  * CF: Could have WGPUColor32 of f32s.  
* CF: No real preference.  
* **Easier to not change. But may think about it some more.**  
* Not hard for Dawn to migrate, with C++ overloads.

**Comment by kainino0x** [comment]
Pretty sure this is not going to cause anything more than tiny differences at the binding level. But finally came up with a deciding factor: since the JS API takes GPUColor here (and therefore `f64`s outside of the range of `f32`) we should really just match JS so we don't introduce any more mismatches we have to deal with. Closing.

BTW I have filed https://github.com/gpuweb/gpuweb/issues/5020 because thinking about this I realized we are actually missing a bunch of minutiae from the JS spec.
